src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+    }
+    temp = head;
+    while (temp != NULL)
+    {
+        if (temp->card.color == RED)
+        {
+            R_count++;
+        }
+        else if (temp->card.color == BLUE)
+        {
+            B_count++;
+        }
+        else if (temp->card.color == GREEN)
+        {
+            G_count++;
+        }
+        else
+        {
+            Y_count++;
+        }
+        temp = temp->next;
+    }
+    if (R_count > B_count && R_count > G_count && R_count > Y_count)
+    {
+        return RED;
+    }
+    else if (B_count > R_count && B_count > G_count && B_count > Y_count)
+    {
+        return BLUE;
+    }
+    else if (G_count > R_count && G_count > B_count && G_count > Y_count)
+    {
+        return GREEN;
+    }
+    else if (Y_count > R_count && Y_count > B_count && Y_count > G_count)
+    {
+        return YELLOW;
+    }
+
+    return 0;
+}
+
+/* function that choose the right card to draw */
+/*
+* @brief: function that choose the right card to draw
+*          if player card # > 1
+*           case 1 : no matched color and no matched number
+*             - if have wild card, draw wild if wild card > with color with the most occurence
+*             - if have skip, use skip
+*             - if not have wild or skip, take card from pile
+*            case 2 : have the matched color and no matched number
+*             - choose the largest number on that color
+*             - if matched color > 3, use match draw/draw2/skip card based on the color with the most occurence
+*
+*            case 3 : no matched color, but have matched number
+*             - if have multiple, choose the color with the most occurence and corresponding action card
+*             - if have only one, get rid of it
+*
+*            case 4 : have matched color and match number
+*             - quickly get rid of matched numbers in the hand first.
+*             - if occurence of
+*
+*         if player card # = 1
+*            - use action card if possibile
+*            - avoid play the last color that the player played
+*
+* @return int the case number
+*/
+
+
+int pick_case(struct CARD card, struct DECK *hand_card)
+{
+    struct DECK *ptr1 = hand_card, *ptr2 = hand_card;
+    int commonColor = 0, commonName = 0;
+    while (ptr1 != NULL) {
+        if (card.color == ptr1->card.color) {
+            commonColor = 1;
+        }
+        ptr1 = ptr1->next;
+    }
+    while (ptr2 != NULL) {
+        if (card.name == ptr2->card.name) {
+            commonName = 1;
+        }
+        ptr2 = ptr2->next;
+    }
+    if (commonColor && commonName) {
+        return 4;
+    }
+    else if (!commonColor && commonName) {
+        return 3;
+    }
+    else if (commonColor && !commonName) {
+        return 2;
+    }
+    else if (!commonColor && !commonName) {
+        return 1;
+    }
+    return 0;
+}
+
+/*
+* @brief: function that play card from handcard list
+*
+* @return void
+*/
+void play_card(struct CARD card, struct DECK **head)
```

Comment:
The return type should be Boolean to check if play_card() is successful or failed.